crates/rpc/src/handlers/validator.rs

```diff
@@ -375,3 +375,83 @@ pub async fn post_validator_balances_from_state(
         ))),
     )
 }
+
+#[derive(Debug, Serialize)]
+pub struct ValidatorLivenessData {  
+    #[serde(with = "serde_utils::quoted_u64")]  
+    pub index: u64,  
+    pub is_live: bool,  
+}
+
+impl ValidatorLivenessData {  
+    pub fn new(index: u64, is_live: bool) -> Self {  
+        Self { index, is_live }  
+    }  
+}  
+
+#[post("/validator/liveness/{epoch}")]  
+pub async fn post_validator_liveness(  
+    db: Data<ReamDB>,  
+    epoch: Path<u64>,  
+    validator_ids: Json<Vec<ValidatorID>>,  
+) -> Result<impl Responder, ApiError> {  
+    let epoch = epoch.into_inner();  
+    let validator_ids = validator_ids.into_inner();  
+      
+    let slot = epoch * SLOTS_PER_EPOCH;  
+    let state = get_state_from_id(ID::Slot(slot), &db).await?;  
+      
+    let mut liveness_data = Vec::new();  
+      
+    for validator_id in validator_ids {  
+        let (index, _validator) = match validator_id {  
+            ValidatorID::Index(i) => {  
+                match state.validators.get(i as usize) {  
+                    Some(validator) => (i as usize, validator),  
+                    None => continue,  
+                }  
+            },  
+            ValidatorID::Address(pubkey) => {  
+                match state.validators.iter().enumerate()  
+                    .find(|(_, v)| v.pubkey == pubkey) {  
+                    Some((i, validator)) => (i, validator),  
+                    None => continue,  
+                }  
+            }  
+        };  
```

Comment:
<img width="467" alt="image" src="https://github.com/user-attachments/assets/06cf1704-bc1f-4f9b-9029-25104114fd9f" />
doesn't the spec say the input will be only indices, not addresses as well?